### PR TITLE
🥔✨ `Journal`: `Entries` link to their `Keywords`

### PR DESCRIPTION
**Pull request by zspencer**
- https://github.com/zinc-collective/convene/issues/1566
- https://github.com/zinc-collective/convene/issues/1662

Screenshot 2023-07-20 at 12 41 16 PM

### CODE REVIEW

File: app/furniture/journal/entry_component.rb [hunk 1]
```diff
@@ -9,10 +9,21 @@ def initialize(*args, entry:, **kwargs)
...
+      postprocess(render_markdown(entry.body))
     end
 
-    def published_at
+    private def postprocess(text)
+      entry.keywords.map do |keyword|
+        search = keyword.canonical_with_aliases.sort.reverse.join("|")
+        text.gsub!(/\#(#{search})/i) do |match|
+          link_to(match, keyword.location)
+        end
```
Comment (rosschapman) [inline app/furniture/journal/entry_component.rb:20]: What about extracting this into a separate (named) helper. Anytime I encounter a regex I find I spend too much time figuring out what it's trying to do.

**Review by rosschapman — APPROVED** [review]

File: app/furniture/journal/entry_component.rb (hunk 1 above)
Comment (zspencer) [reply app/furniture/journal/entry_component.rb:20]: I'm not sure what the name would be; or the right seam... If you have more specific suggestions I'd be down to try something.

Comment (zspencer) [reply app/furniture/journal/entry_component.rb]: I pulled out a few small methods with more helpful names, which I think may help? LMK if not.

File: app/furniture/journal/entry_component.rb [hunk 2]
```diff
@@ -9,10 +9,29 @@ def initialize(*args, entry:, **kwargs)
...
+      text
+    end
+
+    private def replace_keyword(text, keyword)
+      text.gsub!(keywords_regex(keyword)) do |match|
+        link_to(match, keyword.location)
+      end
+    end
+
+    # We sort the keywords longest to shortest because regex matches groups left-to-right
```
Comment (rosschapman) [inline app/furniture/journal/entry_component.rb:29]: Super helpful! ❤️

File: app/furniture/journal/entry_component.rb [hunk 3]
```diff
@@ -9,10 +9,29 @@ def initialize(*args, entry:, **kwargs)
...
-    def published_at
+    private def postprocess(text)
+      entry.keywords.map do |keyword|
+        replace_keyword(text, keyword)
+      end
+
+      text
+    end
+
+    private def replace_keyword(text, keyword)
```
Comment (rosschapman) [inline app/furniture/journal/entry_component.rb:23]: Other name suggestions my brain comes up with: 
- replace_keywords_with_link
- keywords_to_link
- linkify_keywords

☺️

Comment (zspencer) [reply app/furniture/journal/entry_component.rb:23]: I like `linkify_keywords`! Wil try to do that.

**Pull request merged**